nyx_client/nyx_client/client.py

```diff
@@ -599,20 +609,25 @@ def create_data(
             "lang": lang,
             "status": status,
             "preview": preview_base64_string,
-            "downloadURL": download_url,
             "contentType": content_type,
         }
+        if download_url:
+            data["download_url"] = download_url
         if price:
             data["price"] = price
-
         if license_url:
             data["licenseURL"] = license_url
 
-        multipart_data = MultipartEncoder(
-            fields={
-                "productMetadata": json.dumps(data),
-            }
-        )
+        fields = {
+            "productMetadata": json.dumps(data),
+        }
+
+        if file:
+            file.seek(0)
+            # Add file to fields with filename
```

Comment:
(unimportant) comment doesn't clarify much over what is already shown. Maybe makes more sense above the `if file` to point at the fact that this branch handles upload?